libs/runtime/ebpf_epoch.c

```diff
@@ -302,6 +346,20 @@ ebpf_epoch_initiate()
         }
     }
 
+    // Set the initial state for CPU 0.
+    // This code can't use _ebpf_epoch_activate_cpu as it may not running on CPU 0.
```

Comment:
I don't understand this comment.  _ebpf_epoch_activate_cpu() takes the CPU id as an argument.
Why does it matter what it's running on?  If it matters, then the comments on _ebpf_epoch_activate_cpu() are lacking.

Reply:
Thanks. I will fix this. Part of the issue is that activate must run on the target CPU. Calling activate on CPU N from CPU M will cause tearing of some state, so to detect this we assert if it's a cross-cpu activation. But the initialization can run from any CPU and still need to activate CPU. 

Updating this to set thread affinity to run this from CPU 0 in a subsequent iteration.

Reply:
Hope it is clearer now.